crates/checkpoint-downloader/src/downloader.rs

```diff
@@ -558,6 +562,21 @@ fn handle_checkpoint_error(err: &RetriableClientError, next_checkpoint: u64) {
             "failed to read next checkpoint, probably not produced yet",
         );
     }
+
+    if let RetriableClientError::FallbackError(fallback_error) = err
+        && let FallbackError::RequestFailed(error) = fallback_error
+        && error
+            .status()
+            .map(|status| status == axum::http::StatusCode::NOT_FOUND)
+            .unwrap_or(false)
```

Comment:
Nit: Maybe the following is more intuitive?
```suggestion
            .is_some_and(|status| status == axum::http::StatusCode::NOT_FOUND)
```